.github/workflows/docs.yml

```diff
@@ -0,0 +1,194 @@
+        with:
+          service-key: ${{ secrets.DOCS_KEEP_NETWORK_UPLOADER_SERVICE_KEY_JSON }}
+          project: keep-test
+          bucket-name: docs.keep.network
+          bucket-path: core/${{ github.head_ref }}-wip2 # TODO: remove `-wip2` after tests
+          build-folder: tmp/docs
+    
+  docs-solidity:
+    runs-on: ubuntu-latest
+    needs: docs-detect-changes
+    continue-on-error: true
+    if: |
+      github.event_name != 'pull_request'
+        || needs.docs-detect-changes.outputs.path-filter == 'true'
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Generate solidity docs
+        run: |
+          cd solidity && npm install
+          node_modules/.bin/truffle compile
+          mkdir -p ${{ github.workspace }}/docs/solidity
+          node scripts/generate-api-docs.js > ${{ github.workspace }}/docs/solidity/index.adoc
+
+      - name: Upload a Build Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: Solidity Docs
+          path: ${{ github.workspace }}/docs/solidity/index.adoc
+
+  docs-html:
+    runs-on: ubuntu-latest
+    needs: [docs-detect-changes, docs-solidity]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Download a Build Artifact
+        uses: actions/download-artifact@v2
+        continue-on-error: true
+        with:
+          name: Solidity Docs
+          path: ${{ github.workspace }}/docs/solidity
+
+      - name: Build HTML docs
+        id: html
+        uses: thesis/asciidoctor-action@v1.0.1
+        with:
+          files: 'docs/*.adoc docs/**/*.adoc'
+          args: '-a revdate=`date +%Y-%m-%d` --failure-level=ERROR'
+
+      # A push event is a master merge; deploy to primary bucket.
+      - if: github.event_name == 'push'
+        name: Upload asciidocs
+        uses: thesis/gcp-storage-bucket-action@v3.1.0
+        with:
+          service-key: ${{ secrets.DOCS_KEEP_NETWORK_UPLOADER_SERVICE_KEY_JSON }}
+          project: keep-test
+          bucket-name: docs.keep.network
+          bucket-path: core-wip2 # TODO: remove `-wip2` after tests
+          build-folder: ${{ steps.html.outputs.asciidoctor-artifacts }}/docs
+
+      # A pull_request event is a PR; deploy to preview bucket.
+      - if: github.event_name == 'pull_request'
+        name: Upload asciidocs preview
+        uses: thesis/gcp-storage-bucket-action@v3.1.0
+        with:
+          service-key: ${{ secrets.DOCS_KEEP_NETWORK_UPLOADER_SERVICE_KEY_JSON }}
+          project: keep-test
+          bucket-name: docs.keep.network
+          bucket-path: core/${{ github.head_ref }}-wip2 # TODO: remove `-wip2` after tests
+          build-folder: ${{ steps.html.outputs.asciidoctor-artifacts }}/docs
+
+      - name: Upload a Build Artifact # TODO: Remove this after tests finished
+        uses: actions/upload-artifact@v2
+        with:
+          name: HTML Docs
+          path: ${{ steps.html.outputs.asciidoctor-artifacts }}/docs
```

Comment:
We don't need this anymore, right?

Reply:
Right, I checked the content of the artifact for latest build (everything ok) and finally removed the step in 8c33123f4fc6bb176a66fd8bfc69e9a9437d3458.